lib/core/versions/latest/PublicKeyPurpose.ts

```diff
@@ -1,7 +1,7 @@
 /**
- * Sidetree public key verificationRelationship.
+ * DID Document public key purpose.
  */
-enum VerificationRelationship {
+enum PublicKeyPurpose {
   VerificationMethod = 'verificationMethod',
```

Comment:
This should not be commingled as a purpose, as it is not in the DID Core spec. This should be the array all keys are printed in, and then referenced from via IDs.

Reply:
```
{
  "@context": [
    "https://www.w3.org/ns/did/v1",
    {
      "@base": "did:key:z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1"
    }
  ],
  "id": "did:key:z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1",
  "verificationMethod": [
    {
      "id": "#z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1",
      "type": "JsonWebKey2020",
      "controller": "did:key:z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1",
      "publicKeyJwk": {
        "crv": "Ed25519",
        "x": "KKtmCrqQ4EC1B3v6ok6jGJR156NpN0a_5PgZcplxFNw",
        "kty": "OKP"
      }
    },
    {
      "id": "#z6LSrhYgWXdWJcqxniVvTD5Ve85f9E2YpW9PbvNzRZ5VpgBN",
      "type": "JsonWebKey2020",
      "controller": "did:key:z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1",
      "publicKeyJwk": {
        "kty": "OKP",
        "crv": "X25519",
        "x": "3zk2zPWpgich_-x59_DqBujI3Pkk-LQMAghqfwBCt20"
      }
    }
  ],
  "authentication": [
    "#z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1"
  ],
  "assertionMethod": [
    "#z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1"
  ],
  "capabilityInvocation": [
    "#z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1"
  ],
  "capabilityDelegation": [
    "#z6MkhC1wRfHJ6SzxtTpFJGR6S5QTS7hze6B6jFZJsYBbbMf1"
  ],
  "keyAgreement": [
    "#z6LSrhYgWXdWJcqxniVvTD5Ve85f9E2YpW9PbvNzRZ5VpgBN"
  ]
}
```

Reply:
@csuwildcat I think you are confusing `purpose` as a property of the sidetree publickey model, with the concept of verification relationships, which is defined in did core:

https://w3c.github.io/did-core/#verification-relationships

Reply:
`verificationMethod` is the "general" purpose... its totally valid to expose a key that is not "general" in fact its a security anti pattern to list all keys as "general"... its better to delegate keys for specific purpose, and never reuse the same key for multiple purposes.

Reply:
the did:key example I provided can't handle security that way, but sidetree did methods can.

Reply:
also, `verificationMethod` is in the did core spec, its what replaced public key: https://w3c.github.io/did-core/#verification-methods

since public key can't describe verification methods such as bitcoin addresses or hash based proofing mechanisms.

Reply:
I agree with @csuwildcat actually, this will be more broken if we don't do what you are suggesting.